packages/plugin-core/src/utils/ExtensionUtils.ts

```diff
@@ -439,6 +441,52 @@ export class ExtensionUtils {
     if (previewTheme !== undefined) {
       _.set(trackProps, "previewTheme", previewTheme);
     }
+
+    const allExtensions = vscode.extensions.all;
+    let allNonBuiltInExtensions = allExtensions.filter((extension) => {
+      return !extension.packageJSON.isBuiltin;
+    });
+    if (VSCodeUtils.isDevMode()) {
+      // done to make this work during dev mode
+      allNonBuiltInExtensions = allNonBuiltInExtensions.filter((ext) => {
+        return !ext.extensionPath.includes("packages/plugin-core");
+      });
+    }
+    try {
+      const extensionsDetail = allNonBuiltInExtensions.map((extension) => {
+        const { packageJSON } = extension;
+        const { id, version } = packageJSON;
+        return { id, version };
+      });
+      if (extensionsDetail && extensionsDetail.length > 0) {
+        _.set(trackProps, "extensionsDetail", JSON.stringify(extensionsDetail));
```

Comment:
Why `JSON.stringify`? I think this property will need to be an array, else it's going to be difficult to analyze the data in amplitude.

Reply:
to prevent some unfortunate column number blowout in postgres, but that's not going to be the case here since it's an array, and they will be stringified on Segment's end anyways.

I think Amplitude parses it as json on their end too, so the stringify here is unnecessary / excessive.

I'll change it.